change-validation/src/test/scala/com/normation/plugins/changevalidation/UnsupervisedTargetsRepositoryTest.scala

```diff
@@ -0,0 +1,70 @@
+package com.normation.plugins.changevalidation
+
+import better.files.File
+import com.normation.JsonSpecMatcher
+import com.normation.rudder.domain.nodes.NodeGroupId
+import com.normation.rudder.domain.nodes.NodeGroupUid
+import com.normation.rudder.domain.policies.GroupTarget
+import com.normation.rudder.domain.policies.SimpleTarget
+import com.normation.zio.UnsafeRun
+import java.nio.file.Files
+import java.nio.file.attribute.PosixFilePermissions
+import org.junit.runner.RunWith
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+import scala.jdk.CollectionConverters._
+
+@RunWith(classOf[JUnitRunner])
+class UnsupervisedTargetsRepositoryTest extends Specification with JsonSpecMatcher {
+  sequential
+  isolated
+
+  "UnsupervisedTargetsRepository" should {
+    "create the directory and file if it does not exist" in {
+      val tmpDir           = File(Files.createTempDirectory("rudder-test-"))
+      val tmpDirPath       = tmpDir.delete().path
+      val unsupervisedRepo = new UnsupervisedTargetsRepository(tmpDirPath, "unsupervised-targets.json")
+
+      unsupervisedRepo.checkPathAndInitRepos().runNow
+      val file = tmpDir / "unsupervised-targets.json"
+      file.exists must beTrue
+      file.contentAsString must equalsJsonSemantic("""{"unsupervised":[]}""")
+    }
+
+    val tmpDir     = File(Files.createTempDirectory("rudder-test-"))
+    val tmpDirPath = tmpDir.path
+    val file       = tmpDir / "unsupervised-targets.json"
+
+    "do nothing if the directory is not writable" in {
+      val unsupervisedRepo = new UnsupervisedTargetsRepository(tmpDirPath, "unsupervised-targets.json")
+      tmpDir.setPermissions(PosixFilePermissions.fromString("r--r--r--").asScala.toSet)
```

Comment:
this need to be at least `rw-r--r--` else the test modifying it fails.  (`SupervisedTargetsApiTest`, the `POST`)

Reply:
That's right the tests were not passing ! The change was there : https://github.com/Normation/rudder-plugins/blob/31df6b8475ea9c5ae8e6de8e31c93a6b213ef2e0/change-validation/src/test/scala/com/normation/plugins/changevalidation/api/SupervisedTargetsApiTest.scala#L28 (I think I changed it just to see the response I forgot to revert back to `rw` :smile:)